lxd/networks_utils.go

```diff
@@ -1132,3 +1195,121 @@ func networkGetState(netIf net.Interface) api.NetworkState {
 	network.Counters = shared.NetworkGetCounters(netIf.Name)
 	return network
 }
+
+func networkConfigureIPVlanAddress(address, parentDevName string) error {
+
+	if address == "" {
+		return fmt.Errorf("Network address can not be empty for parent nic: %s", parentDevName)
+	}
+
+	// handle IPv4
+	if err := networkValidAddressV4(address); err == nil {
+		// add Local Route
+		_, err := shared.RunCommand("ip", "-4", "route", "add", "local", fmt.Sprintf("%s/32", address), "dev", parentDevName)
+		if err != nil {
+			switch {
+			case strings.Contains(err.Error(), "File exists"):
+			default:
+				return err
+			}
+		}
+		// add Proxy ARP record
+		_, err = shared.RunCommand("ip", "-4", "neigh", "add", "proxy", address, "dev", parentDevName)
+		if err != nil {
+			return err
+		}
+	}
+
+	// handle IPv6
+	if err := networkValidAddressV6(address); err == nil {
+		// add Local Route
+		_, err := shared.RunCommand("ip", "-6", "route", "add", "local", fmt.Sprintf("%s/128", address), "dev", parentDevName)
+		if err != nil {
+			switch {
+			case strings.Contains(err.Error(), "File exists"):
+			default:
+				return err
+			}
+		}
+		// add Proxy NDP record
+		_, err = shared.RunCommand("ip", "-6", "neigh", "add", "proxy", address, "dev", parentDevName)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func networkUnConfigureIPVlanAddress(address, parentDevName string) error {
+
+	if address == "" {
+		return fmt.Errorf("Network address can not be empty for parent nic: %s", parentDevName)
+	}
+
+	// handle IPv4
+	if err := networkValidAddressV4(address); err == nil {
+		// delete Local Route
+		_, err := shared.RunCommand("ip", "-4", "route", "del", "local", fmt.Sprintf("%s/32", address), "dev", parentDevName)
+		if err != nil {
+			switch {
+			case strings.Contains(err.Error(), "No such process"):
+			case strings.Contains(err.Error(), "No such file or directory"):
+			default:
+				return err
+			}
+		}
+		// delete Proxy ARP record
+		_, err = shared.RunCommand("ip", "-4", "neigh", "del", "proxy", address, "dev", parentDevName)
+		if err != nil {
+			switch {
+			case strings.Contains(err.Error(), "No such process"):
+			case strings.Contains(err.Error(), "No such file or directory"):
+			default:
+				return err
+			}
+		}
+	}
+
+	// handle IPv6
+	if err := networkValidAddressV6(address); err == nil {
+		// delete Local Route
+		_, err := shared.RunCommand("ip", "-6", "route", "del", "local", fmt.Sprintf("%s/128", address), "dev", parentDevName)
+		if err != nil {
```

Comment:
For those kind of actions, it's usually fine to just completely disregard any error.
Trying to be clever and parse output isn't going to work so well on systems that are using a different locale.

Reply:
Same applies to the other few one of those.

Reply:
disregarding an error feels like a hack, if something goes wrong, debugging would be harder.

I purpose changing a bit `shared.RunCommand` so it will include exit code inside error message if exit code is non-zero.

Can be done in separate PR, what do you think?

Reply:
something like this maybe https://github.com/lxc/lxd/pull/5225 ?
@stgraber 
and then switch on exitcode

Reply:
You don't actually need that PR, look for `RunError` in the codebase for how to extract the exit code from a RunCommand call.

Reply:
fixed in https://github.com/s3rj1k/lxd/commit/81052632dd2dc5393d566c280da3eb585b10c538